### PR TITLE
Added Service Endpoints cmdlets

### PR DESCRIPTION
**Pull request by EvgenyAgafonchikov**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by DeepakRajendranMsft** [comment]
@cormacpayne, @markcowl  could you please take a look at this.

We have updated nrp api version and reintroduced the private access feature.

we have other cmdlets incoming so would like to get this in so the CR becomes smaller

**Comment by cormacpayne** [comment]
@azuresdkci test this please

File: src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj [hunk 1]
```diff
@@ -82,10 +82,22 @@
       <HintPath>..\..\..\packages\Microsoft.Data.Services.Client.5.8.2\lib\net40\Microsoft.Data.Services.Client.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj]: @EvgenyAgafonchikov you shouldn't need to add these references in your csproj file, they should be derived from [`Common.Dependencies.targets`](https://github.com/Azure/azure-powershell/blob/preview/tools/Common.Dependencies.targets)

File: src/ResourceManager/Compute/Commands.Compute/packages.config [hunk 2]
```diff
@@ -3,7 +3,10 @@
   <package id="AutoMapper" version="3.1.1" targetFramework="net45" />
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Compute" version="16.2.0" targetFramework="net452" />
-  <package id="Microsoft.Azure.Management.Network" version="12.0.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.Network" version="14.0.0-preview" targetFramework="net452" />
   <package id="Microsoft.Azure.Management.Storage" version="4.1.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.9" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/packages.config]: @EvgenyAgafonchikov same comment, you shouldn't need to add the references in this file, they should be derived from the [`packages.config`](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.ResourceManager.Common/packages.config) file in `Commands.ResourceManager.Common`

File: src/ResourceManager/Network/ChangeLog.md [hunk 3]
```diff
@@ -30,6 +30,12 @@
     - Add-AzureRmApplicationGatewayBackendHttpSettings
     - New-AzureRmApplicationGatewayBackendHttpSettings
     - Set-AzureRmApplicationGatewayBackendHttpSettings 
+* Added support for endpoint services to Virtual Network Subnets
```
Comment (cormacpayne) [inline src/ResourceManager/Network/ChangeLog.md]: @EvgenyAgafonchikov these changes should go under `Current Release`

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 4]
```diff
@@ -53,9 +53,20 @@
...
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=12.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.12.0.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=14.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.14.0.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: @EvgenyAgafonchikov same comment about removing the references added here

File: src/ResourceManager/Network/Commands.Network/packages.config [hunk 5]
```diff
@@ -2,5 +2,8 @@
 <packages>
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Compute" version="16.0.0" targetFramework="net452" />
-  <package id="Microsoft.Azure.Management.Network" version="12.0.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.Network" version="14.0.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.9" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/packages.config]: @EvgenyAgafonchikov same comment about removing the added references

**Comment by DeepakRajendranMsft** [comment]
@cormacpayne  are there any pending items that is preventing merge here?

Thanks,
deepak.

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/job/powershell-demand/1856/

**Pull request merged**